src/repositories/AuxiliaryChainRepository.ts

```diff
@@ -141,98 +118,83 @@ export class AuxiliaryChainRepository { // extends Subject<AuxiliaryChain> {
       },
       {
         ...initOptions,
-        modelName: 'auxiliaryChain',
-        tableName: 'auxiliary_chain',
+        modelName: 'AuxiliaryChain',
+        tableName: 'auxiliary_chains',
       },
     );
   }
 
   /**
-   * Creates an auxiliary chain model in the repository and syncs with database.
-   * @param {AuxiliaryChainAttributes} auxiliaryChainAttributes
-   * @return {Promise<AuxiliaryChain>}
+   * Saves a AuxiliaryChain object in the repository.
+   * If a AuxiliaryChain does not exist, it creates, otherwise updates.
+   *
+   * @param auxiliaryChain AuxiliaryChain object.
+   *
+   * @returns Newly created or updated AuxiliaryChain object.
    */
-  public async create(auxiliaryChainAttributes: AuxiliaryChainAttributes): Promise<AuxiliaryChain> {
-    try {
-      const auxiliaryChain: AuxiliaryChain = await AuxiliaryChainModel.create(
-        auxiliaryChainAttributes,
-      ) as AuxiliaryChain;
-      this.format(auxiliaryChain);
-      // this.newUpdate(auxiliaryChain);
-      return auxiliaryChain;
-    } catch (e) {
-      const errorContext = {
-        attributes: auxiliaryChainAttributes,
-        reason: e.message,
-      };
-      return Promise.reject(
-        `Failed to create an auxiliary chain: ${JSON.stringify(errorContext)}`,
-      );
-    }
+  public async save(auxiliaryChain: AuxiliaryChain): Promise<AuxiliaryChain> {
+    const definedOwnProps: string[] = Utils.getDefinedOwnProps(auxiliaryChain);
+
+    await AuxiliaryChainModel.upsert(
+      auxiliaryChain,
+      {
+        fields: definedOwnProps,
+      },
+    );
+
+    const updatedAuxiliaryChain = await this.get(
+      auxiliaryChain.chainId,
+    );
```

Comment:
Please, assert that result is not null as in the next lines we cast.
I've checked in StakeRequestRepository::save() method. There we assert that it's not undefined, however, correct assertion is that it's not null.

Reply:
Done!